lib/puppet/indirector/request.rb

```diff
@@ -42,7 +42,8 @@ def environment=(env)
   end
 
   def escaped_key
-    URI.escape(key)
+    # TODO: these keys are generally fragments so Puppet::Util.uri_escape isn't really appropriate
+    Puppet::Util.uri_escape(key)
   end
```

Comment:
I was just looking into this... the only caller to this method is at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/http/api/indirected_routes.rb#L228 which means it always appears in the path of the URI.

I think I may remove this method and inline it into the single callsite...